crates/stages/src/error.rs

```diff
@@ -20,6 +20,15 @@ pub enum StageError {
         #[source]
         error: consensus::ConsensusError,
     },
+    /// The stage encountered a downloader error where the responses cannot be attached to the
+    /// current head.
+    #[error("Stage encountered detached head #{number} ({hash:?})", number = local_head.number, hash = local_head.hash)]
+    DetachedHead {
+        /// The local head header.
+        local_head: SealedHeader,
+        /// The unwind target block number.
+        unwind_to: BlockNumber,
```

Comment:
see other comment.

I'd like to leave this up the pipeline caller what to do in this situation.

should this error also include the "correct" downloaded header?